store/v2/root/store.go

```diff
@@ -230,13 +230,50 @@ func (s *Store) LoadVersion(version uint64) error {
 		defer s.telemetry.MeasureSince(now, "root_store", "load_version")
 	}
 
-	return s.loadVersion(version)
+	return s.loadVersion(version, nil)
 }
 
-func (s *Store) loadVersion(v uint64) error {
+// LoadVersionAndUpgrade implements the UpgradeableRootStore interface.
+//
+// NOTE: It cannot be called while the store is migrating.
+func (s *Store) LoadVersionAndUpgrade(version uint64, upgrades *corestore.StoreUpgrades) error {
+	if s.telemetry != nil {
+		now := time.Now()
+		defer s.telemetry.MeasureSince(now, "root_store", "load_version_and_upgrade")
+	}
+
+	if s.isMigrating {
+		return fmt.Errorf("cannot upgrade while migrating")
+	}
+
+	if err := s.loadVersion(version, upgrades); err != nil {
+		return err
+	}
+
+	// if upgrades is not nil, we need to add the pruned KVStores to the pruning manager
+	if upgrades != nil {
+		kvStoreGetter, ok := s.stateCommitment.(store.KVStoreGetter)
+		if !ok {
```

Comment:
Should this be `ok` without `!` ?  The `kvStoreGetter` is used inside.

Should there be an error or info log for the else cause?